public/app/plugins/datasource/loki/syntax.ts

```diff
@@ -192,7 +192,7 @@ export const lokiGrammar: Grammar = {
         pattern: /#.*/,
       },
       'label-key': {
-        pattern: /[a-z_]\w*(?=\s*(=|!=|=~|!~))/,
+        pattern: /[a-z_,A-Z_]\w*(?=\s*(=|!=|=~|!~))/,
```

Comment:
I think the comma should be removed as well as the redundant second underscore. Including the comma makes it a valid option in the regex and I think you have have been using it like an 'or' which the brackets already do. 

Here's the Prometheus label data model which includes a regex for valid labels as reference: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels